R-package/tests/testthat/test_dataset.R

```diff
@@ -85,6 +85,8 @@ test_that("lgb.Dataset: Dataset should be able to construct from matrix and retu
     , ref_handle
   )
   expect_false(is.na(handle))
+  lgb.call("LGBM_DatasetFree_R", ret = NULL, handle)
```

Comment:
I suggest to avoid all `handle`s in tests. The leakage in tests is hard to trace.